### PR TITLE
Adding the steps to build CHIP for Android apps.

### PR DESCRIPTION
**Pull request by vidhis88**

### CODE REVIEW

File: src/android/CHIPTool/README.md [hunk 1]
```diff
@@ -6,3 +6,63 @@ a CHIP device
...
+
+Steps to build CHIP for Android apps
+
+Pre-conditions: Have Android NDK downloaded to your machine. Set $ANDROID_NDK_HOME environment variable to point to the NDK package is downloaded.
+
+1. Checkout the CHIP repo
+
+2. In commandline / Terminal, 'cd' into the top CHIP directory
+
+3. May have to run 'autoconf' / 'autoreconf', if changes to configure.ac have been made.
```
Comment (gerickson) [inline src/android/CHIPTool/README.md:20]: This should be `./bootstrap -w config` or just `./bootstrap`.

**Review by gerickson — APPROVED** [review]
Minor tweak requested.

**Review by BroderickCarlin — APPROVED** [review]

File: src/android/CHIPTool/README.md [hunk 2]
```diff
@@ -6,3 +6,63 @@ a CHIP device
...
+5. Exit out of the checked out CHIP repo directory and using 'mkdir [some-dir]' create
+a new empty one where you want the generated output files to go.
+
+6. 'cd' into this newly created directory
+
+7. Export the following variables
+
+ABIs that can be used as TARGET
+    armeabi-v7a:  armv7a-linux-androideabi
+    arm64-v8a:	  arch64-linux-android
```
Comment (readmodifywrite) [inline src/android/CHIPTool/README.md:33]: Should be aarch64-linux-android

**Review by woody-apple — APPROVED** [review]

File: src/android/CHIPTool/README.md [hunk 3]
```diff
@@ -6,3 +6,63 @@ a CHIP device
...
+export AS=$TOOLCHAIN/bin/$TARGET-as
+export CC=$TOOLCHAIN/bin/$TARGET$API-clang
+export CXX=$TOOLCHAIN/bin/$TARGET$API-clang++
+export LD=$TOOLCHAIN/bin/$TARGET-ld
+export RANLIB=$TOOLCHAIN/bin/$TARGET-ranlib
+export STRIP=$TOOLCHAIN/bin/$TARGET-strip
+
+8. Then run '../connectedhomeip/configure --host=$TARGET --with-crypto=mbedtls --enable-tests=no --enable-shared'
+
+9. Finally 'make'
```
Comment (readmodifywrite) [inline src/android/CHIPTool/README.md:56]: There's an issue with a Java include path here.

In file included from ../../../../connectedhomeip/src/setup_payload/java/SetupPayloadParser-JNI.cpp:4:
/Library/Java/JavaVirtualMachines/jdk-12.0.1.jdk/Contents/Home/include/jni.h:45:10: fatal error: 'jni_md.h' file not found
#include "jni_md.h"
         ^~~~~~~~~~

Comment (saurabhst) [reply src/android/CHIPTool/README.md:56]: @vidhis88 ^

**Review by hawk248 — APPROVED** [review]

File: src/android/CHIPTool/README.md (hunk 1 above)
Comment (vidhis88) [reply src/android/CHIPTool/README.md]: Done.

File: src/android/CHIPTool/README.md (hunk 2 above)
Comment (vidhis88) [reply src/android/CHIPTool/README.md]: Thanks for catching the typo. Done.

**Review by saurabhst — APPROVED** [review]

**Review by robszewczyk — APPROVED** [review]

File: src/android/CHIPTool/README.md (hunk 3 above)
Comment (vidhis88) [reply src/android/CHIPTool/README.md]: Updated the README to include:
JNI_INCLUDE_DIRS=$ANDROID_NDK_HOME/sysroot/usr/include
to fix this error.

**Comment by saurabhst** [comment]
Is this ready to be merged?

**Comment by woody-apple** [comment]
> Is this ready to be merged?

Needs restyle to be merged in, I can do that for @vidhis88 in a few minutes

**Review by jelderton — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Comment by vidhis88** [comment]
> > Is this ready to be merged?
> 
> Needs restyle to be merged in, I can do that for @vidhis88 in a few minutes

I just updated my PR with the restyler suggestions.
Should be ready to merge.

**Comment by woody-apple** [comment]
I'll bring them in again - It didn't like the suggestions, you can in future just follow the instructions in the linked PR (has command line instructions)

**Pull request merged**